Framework/Crystal/src/PeakStatisticsTools.cpp

```diff
@@ -0,0 +1,330 @@
+#include "MantidCrystal/PeakStatisticsTools.h"
+
+#include "MantidGeometry/Crystal/BasicHKLFilters.h"
+#include "MantidGeometry/Crystal/HKLGenerator.h"
+
+#include "MantidKernel/Statistics.h"
+
+#include <boost/make_shared.hpp>
+#include <numeric>
+
+namespace Mantid {
+namespace Crystal {
+namespace PeakStatisticsTools {
+
+using namespace Mantid::DataObjects;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+/// Returns a vector with the intensities of the Peaks stored in this
+/// reflection.
+std::vector<double> UniqueReflection::getIntensities() const {
+  std::vector<double> intensities;
+  intensities.reserve(m_peaks.size());
+
+  std::transform(
+      m_peaks.begin(), m_peaks.end(), std::back_inserter(intensities),
+      [](const DataObjects::Peak &peak) { return peak.getIntensity(); });
+
+  return intensities;
+}
+
+/// Returns a vector with the intensity sigmas of the Peaks stored in this
+/// reflection.
+std::vector<double> UniqueReflection::getSigmas() const {
+  std::vector<double> sigmas;
+  sigmas.reserve(m_peaks.size());
+
+  std::transform(
+      m_peaks.begin(), m_peaks.end(), std::back_inserter(sigmas),
+      [](const DataObjects::Peak &peak) { return peak.getSigmaIntensity(); });
+
+  return sigmas;
+}
+
+/// Removes peaks whose intensity deviates more than sigmaCritical from the
+/// intensities' mean.
+void UniqueReflection::removeOutliers(double sigmaCritical) {
+  if (sigmaCritical <= 0.0) {
+    throw std::invalid_argument(
+        "Critical sigma value has to be greater than 0.");
+  }
+
+  if (m_peaks.size() > 2) {
```

Comment:
I changed this entire thing to return a new `UniqueReflection` instead. That also solves the problem with the `const` map in the calculation function and it's generally nicer to not change state. Maybe the `setIntensities` thing should work the same way.